src/Action.php

```diff
@@ -60,6 +60,28 @@ public function __construct(BaseAction &$action)
     }
 
     /**
+     * Returns the path for an action source file.
+     *
+     * @param string $filename
+     * @return string
+     * @throws \Exception
+     */
+    private function getSourcePath(string $filename): string
+    {
+        $path = implode(DIRECTORY_SEPARATOR, [
+            dirname($_SERVER['SCRIPT_NAME']),
+            'actions',
+            "$filename.php"
+        ]);
+
+        if (!is_readable($path)) {
+            throw new \Exception('Source file not readable');
```

Comment:
The problem is that if the `Exception` is not caught it will terminate the execution of the action.

Could it be this:

```php
$path = $this->getSourcePath($this->action->getActionName());
if ($path) {
    $callback = require $path;
    // do something with $callback
}
return $this;
```